FindNewPython's forces check against PYTHON_EXECUTABLE during configuration phase

A recent change to pybind11 does [checks against a python interpreter](https://github.com/pybind/pybind11/commit/928fff649fa52ebc7b43860a4914f68e320d2bb2 ). The verification done for FindPythonLibs doesn’t apply in the case of [superbuilt python](https://github.com/InsightSoftwareConsortium/ITKExamples/blob/master/Superbuild/External-Python.cmake) (ExternalProject_Add) since the PYTHON_EXECUTABLE isn't ready until compile time.

In my circumstances, I already have ``${PYTHON_LIBRARY}`` and ``${PYTHON_INCLUDE_DIR}`` set.  CMake's [FindPythonLibs](https://github.com/Kitware/CMake/blob/master/Modules/FindPythonLibs.cmake) allows you to specify your versions:

> If you'd like to specify the installation of Python to use, you should
modify the following cache variables:

> ```
>      PYTHON_LIBRARY             - path to the python library
>      PYTHON_INCLUDE_DIR         - path to where Python.h is found
> ```

Expected behavior: ``FindPythonLibsNew.cmake`` should skip ``execute_process(COMMAND "${PYTHON_EXECUTABLE}" "-c"`` and/or fallback to ``find_package(PythonLibs)`` behavior if ``PYTHON_LIBRARY`` and ``PYTHON_INCLUDE_DIR`` is already set.

@dean0x7d 

During ``add_subdirectory(pybind11)``, I get the same with and without ``pybind11_add_module``.

```
❯ cmake ..  -GNinja -DCMAKE_BUILD_TYPE=Debug
-- WITH_PYMALLOC
-- PYTHON_INCLUDE_DIR: /Users/me/study/c++/kakoune/build/superbuild/Python-install/include/python3.5
-- PYTHON_LIBRARY: /Users/me/study/c++/kakoune/build/superbuild/Python-install/lib/libpython3.5m.dylib
CMake Error at pybind11/tools/FindPythonLibsNew.cmake:93 (message):
  Python config failure:

Call Stack (most recent call first):
  pybind11/CMakeLists.txt:26 (find_package)


-- Configuring incomplete, errors occurred!
See also "/Users/me/study/c++/kakoune/build/CMakeFiles/CMakeOutput.log".
```

@wjakob @dean0x7d  I'm preparing more details

--- a/tools/FindPythonLibsNew.cmake
+++ b/tools/FindPythonLibsNew.cmake
@@ -7,13 +7,10 @@
 #  PYTHON_PREFIX              - path to the Python installation
 #  PYTHON_LIBRARIES           - path to the python library
 #  PYTHON_INCLUDE_DIRS        - path to where Python.h is found
+#  PYTHON_MODULE_EXTENSION    - lib extension, e.g. '.so' or '.pyd'
+#  PYTHON_MODULE_PREFIX       - lib name prefix: usually an empty string
 #  PYTHON_SITE_PACKAGES       - path to installation site-packages
 #  PYTHON_IS_DEBUG            - whether the Python interpreter is a debug build
-#
-#  PYTHON_INCLUDE_PATH        - path to where Python.h is found (deprecated)
-#
-# A function PYTHON_ADD_MODULE(<name> src1 src2 ... srcN) is defined
-# to build modules for python.
 #
 # Thanks to talljimbo for the patch adding the 'LDVERSION' config
 # variable usage.
@@ -53,6 +50,9 @@
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #=============================================================================
 
+if(PYTHONLIBS_FOUND)
+    return()
+endif()
 
 # Use the Python interpreter to find the libs.
 if(PythonLibsNew_FIND_REQUIRED)
@@ -164,10 +164,6 @@ else()
         set(PYTHON_LIBRARY python${PYTHON_LIBRARY_SUFFIX})
     endif()
 endif()
-
-# For backward compatibility, set PYTHON_INCLUDE_PATH, but make it internal.
-SET(PYTHON_INCLUDE_PATH "${PYTHON_INCLUDE_DIR}" CACHE INTERNAL
-          "Path to where Python.h is found (deprecated)")
 
 MARK_AS_ADVANCED(
   PYTHON_LIBRARY

